rabbit-escape-ui-swing/src/rabbitescape/ui/swing/SwingGameLaunch.java

```diff
@@ -381,6 +387,7 @@ public int addToken( int tileX, int tileY, Token.Type ability )
         {
             graphics.playSound( "place_token" );
         }
+        solutionRecorder.append( new PlaceTokenAction( tileX, tileY ) );
         return now;
```

Comment:
But why are these 2 actions happening in different classes? I don't think it's a fault in your code but I am curious.